CMP1903_A1_2324/Testing.cs

```diff
@@ -5,7 +5,52 @@ namespace Dice2
 {
 	public class Testing
 	{
-		
+		public Testing()
+		{
+		}
+
+		//Tests that the Die runs
+        internal void Run()
+        {
+			CreateDie();
+			TestRollDie();
+        }
+
+		//Checks whether the Die is null or not
+		public void CreateDie()
+		{
+			Die testdie = new Die();
+			if(testdie == null)
+			{
+				throw new ArgumentNullException();
+			}
+		}
+		//Unit Test - Testing the Die Roll does not have the value 0.
+		public void TestRollDie()
+		{
+			//Arrange
+			Die testdie = new Die();
+			//Act
+			var result = testdie.Rolls();
+			//Assert
+            Debug.Assert(result != 0);
```

Comment:
You've checked that the dice roll doesn't equal 0 which aids in exception handling.